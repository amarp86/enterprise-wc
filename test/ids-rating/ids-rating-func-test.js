/**
 * @jest-environment jsdom
 */

 import IdsRating from '../../src/ids-rating/ids-rating';

 describe('IdsRating Component', () => {
    beforeEach(async () => {
        const ratingComponent = new IdsRating();
        window.document.body.appendChild(ratingComponent)
    });

    afterEach(async () => {
        window.document.body.innerHTML = '';
    });

    it('renders with no errors', () => {
        const errors = jest.spyOn(global.console, 'error');
        const elem = new IdsRating()
        window.document.body.appendChild(elem)
        elem.remove();
        expect(document.querySelectorAll('ids-rating').length).toEqual(1);
        expect(errors).not.toHaveBeenCalled();
    });

    it('renders correctly', () => {
        const elem = new IdsRating();
        window.document.body.appendChild(elem);
        elem.template();
        expect(elem.outerHTML).toMatchSnapshot();
    });

    it('renders rating item correctly', () => {
        const elem = new IdsRating();
        window.document.body.appendChild(elem);
        elem.template();
        const ratingItem = window.document.querySelectorAll('.rating-item');
        console.log(elem)
        //expect(ratingItem).toEqual(5)
    });
 })