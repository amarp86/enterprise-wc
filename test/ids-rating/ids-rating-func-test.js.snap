// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdsRating Component renders correctly 1`] = `"<ids-rating></ids-rating>"`;
